lib/impress.js

```diff
@@ -163,24 +164,22 @@ impress.compareHosts = () => {
   const cmp = [];
   for (appName in impress.applications) {
     config = impress.applications[appName].config;
-    if (config) {
-      hosts = config.hosts;
-      if (hosts) {
-        for (i = 0, ilen = hosts.length; i < ilen; i++) {
-          for (j = 0, jlen = cmp.length; j < jlen; j++) {
-            if (impress.compareMasks(hosts[i], cmp[j])) {
-              impress.log.warning(
-                'Hosts mask overlapping: "' + hosts[i] +
-                '" and "' + cmp[j] + '"'
-              );
-            }
-          }
-        }
-        for (k = 0, klen = hosts.length; k < klen; k++) {
-          if (!cmp.includes(hosts[k])) cmp.push(hosts[k]);
+    if (!config) continue;
+    hosts = config.hosts;
+    if (!hosts) continue;
+    for (i = 0, ilen = hosts.length; i < ilen; i++) {
+      for (j = 0, jlen = cmp.length; j < jlen; j++) {
+        if (impress.compareMasks(hosts[i], cmp[j])) {
+          impress.log.warning(
+            'Hosts mask overlapping: "' + hosts[i] +
+            '" and "' + cmp[j] + '"'
+          );
         }
       }
     }
+    for (k = 0, klen = hosts.length; k < klen; k++) {
```

Comment:
Extra `O(n^2)` loop. It can be done simultaneously with the previous check, like

```js
   for (i = 0, ilen = hosts.length; i < ilen; i++) {
     let hostFound = false;
     for (j = 0, jlen = cmp.length; j < jlen; j++) {
       hostFound = hostFound || (hosts[i] === cmp[j]);
       if (impress.compareMasks(hosts[i], cmp[j])) {
         impress.log.warning(
           'Hosts mask overlapping: "' + hosts[i] +
           '" and "' + cmp[j] + '"'
         );
       }
     }
     if (!hostFound) cmp.push(hosts[i]);
   }
```